src/org/labkey/test/tests/FileAttachmentColumnTest.java

```diff
@@ -15,153 +15,609 @@
 
-        PortalHelper portalHelper = new PortalHelper(getDriver());
-        portalHelper.addWebPart("Sample Types");
+        for (File file : SAMPLE_FILES)
+        {
+            Map<String, String> fileRow = Map.of("Name", file.getName(), "File", file.getAbsolutePath());
+            listHelper.insertNewRow(fileRow, false);
+        }
+    }
+
+    private void importSampleDataUI(String sampleTypeName, String containerPath, List<File> files)
+    {
+        var helper = SampleTypeHelper.beginAtSampleTypesList(this, containerPath);
+        clickAndWait(Locator.linkWithText(sampleTypeName));
+
+        List<Map<String, String>> sampleFileData = new ArrayList<>();
+        for (File file : files)
+        {
+            sampleFileData.add(Map.of("Name", file.getName(), "Color", "green",
+                    "File", file.getName()));
+        }
+        helper.bulkImport(sampleFileData);
+    }
+
+    private Protocol makeGeneralAssay(String assayName, List<PropertyDescriptor> runFields, List<PropertyDescriptor> dataFields,
+                                      String folderPath) throws Exception
+    {
+        var assayDesign =  new GeneralAssayDesign(assayName)
+                .setBatchFields(List.of(new FieldDefinition("batchData", FieldDefinition.ColumnType.String)), false)
+                .setRunFields(runFields, false)
+                .setDataFields(dataFields, false);
+
+
+        var protocol = assayDesign.createAssay(folderPath, createDefaultConnection());
+        var updateProtocol = protocol.setEditableRuns(true).setEditableResults(true);
+        return assayDesign.updateProtocol(folderPath, createDefaultConnection(), updateProtocol);
+    }
+
+    private void addRunData(Integer protocolId, String folderPath) throws Exception
+    {
+        var resultFiles = new ArrayList<File>();
+        List<Map<String, Object>> importData = new ArrayList<>();
+        var uploadHelper = new WebDavUploadHelper(folderPath);  // put the files in the folder root so assay can resolve them
+        for (int i = 0; i < 5; i++)
+        {
+            String fileName = String.format("results_file-%d.tsv", i);
+            String result = String.format("result-%d", i);
+            File otherResultFile = SAMPLE_FILES.get(i);
+            String fileText = "resultTxt\tresultFile\totherResultFile\n"+
+                    result+"\t" + fileName + "\t"+ otherResultFile.getName();
+            var resultFile = TestFileUtils.writeTempFile(fileName, fileText);
+            uploadHelper.uploadFile(resultFile);
+            resultFiles.add(resultFile);
+
+            importData.add(Map.of(RESULT_TXT_COL, result, RESULT_FILE_COL, resultFile.getName(),
+                    OTHER_RESULT_FILE_COL, otherResultFile.getName(),
+                    RUN_TXT_COL,"run text", RUN_FILE_COL, "runFile.tsv"));
+        }
+
+        // generate a run file, referencing the result files
+        String importDataFileContents = TestDataUtils.tsvStringFromRowMaps(importData,
```

Comment:
Could use any file for run file, the content doesn't seem to matter, but using an existing static file should simplify the test

Reply:
OK, I will modify the test accordingly